salt/modules/file.py

```diff
@@ -1651,8 +1651,13 @@ def replace(path,
         with salt.utils.fopen(path,
                               mode='r',
                               buffering=bufsize) as r_file:
+            count_replaced = 0
             for line in r_file:
-                result, nrepl = re.subn(cpattern, repl, line, count)
+                if count == 0 or count_replaced != count:
```

Comment:
Ahh, should that be `count_replaced < count`? My concern is if `nrepl` in the next line is greater than one due to multiple occurrences in one line, then there's a chance `count_replaced` will become greater than `count`, and this loop will keep going.